Library/Homebrew/os/mac.rb

```diff
@@ -120,9 +120,16 @@ def sdk(v = nil)
       sdk_locator.sdk_if_applicable(v)
     end
 
-    def sdk_for_formula(f, v = nil)
+    def sdk_for_formula(f, v = nil, check_only_runtime_requirements: false)
       # If the formula requires Xcode, don't return the CLT SDK
-      return Xcode.sdk if f.requirements.any? { |req| req.is_a? XcodeRequirement }
+      # If check_only_runtime_requirements is true, don't necessarily return the
+      # Xcode SDK if the XcodeRequirement is only a build or test requirment.
+      return Xcode.sdk if f.requirements.any? do |req|
+        next false unless req.is_a? XcodeRequirement
+        next false if check_only_runtime_requirements && !req.required? && !req.test?
```

Comment:
Not sure I understand this logic, can you explain? `required?` already checks `test?`

Reply:
There are a few cases:
|| `req.required?` | `!req.test?` | outcome |
|---|---|---|---|
| `depends_on :xcode` | ❌  | ✅ | ❌ |
| `depends_on xcode: :build` | ✅ | ✅ | ✅ |
| `depends_on xcode: :test` | ✅ | ❌ | ❌ |
| `depends_on xcode: [:build, :test]` | ✅ | ❌ | ❌ |

Does this help to clarify? `required?` essentially means "required at runtime" so it excludes build and test requirements.

Reply:
I think the alternative is to say:

```ruby
next false if check_only_runtime_requirements && (req.test? || req.build?) && !req.test?
```

Ah, now that I'm looking at this, I think this should work:

```ruby
next false if check_only_runtime_requirements && req.build? && !req.test?
```